packages/shared/topology/sidebar/TopologySideBarContent.tsx

```diff
@@ -0,0 +1,107 @@
+import useAlerts from '@odf/shared/monitoring/useAlert';
+import {
+  Alert,
+  K8sResourceCommon,
+  useK8sModel,
+} from '@openshift-console/dynamic-plugin-sdk';
+import * as _ from 'lodash-es';
+import { K8sResourceKind, NodeKind } from '../../types';
+import { useCustomTranslation } from '../../useCustomTranslationHook';
+import Tabs, { TabPage } from '../../utils/Tabs';
+import NodeDetails from './node/NodeDetails';
+import NodeObserve from './node/NodeObserve';
+import NodeResources from './node/NodeResources';
+import './topology-sidebar-content.scss';
+
+type TopologySideBarContentProps = {
+  resource: K8sResourceCommon;
+  getResourceStatus?: (resource: K8sResourceKind) => string;
+};
+
+const TopologySideBarContent: React.FC<TopologySideBarContentProps> = ({
+  resource,
+  getResourceStatus,
+}) => {
+  const { t } = useCustomTranslation();
+  const [model] = useK8sModel(resource.kind);
+  const [alerts, loaded, loadError] = useAlerts();
+  const [alertsTab, detailsTab, resourcesTab, observeTab]: JSX.Element[] =
+    React.useMemo(() => {
+      let alertsFilter = null;
+      let [detailsTab, resourcesTab, observeTab] = Array(4)
+        .fill(null)
+        .map(() => <span>N/A</span>);
+      switch (resource.kind) {
+        case 'Node':
+          alertsFilter = (alert: Alert) =>
+            _.get(alert, 'labels.node', '') === resource.metadata.name ||
+            _.get(alert, 'annotations.description', '').includes(
+              resource.metadata.name
+            ) ||
+            _.get(alert, 'annotations.message', '').includes(
+              resource.metadata.name
+            );
+          detailsTab = <NodeDetails node={resource as NodeKind} />;
+          resourcesTab = <NodeResources node={resource as NodeKind} />;
+          observeTab = <NodeObserve node={resource as NodeKind} />;
+          break;
+      }
+      const alertsTab = (
+        <AlertsPanel
+          alerts={alerts}
+          alertsFilter={alertsFilter}
+          loaded={loaded}
+          loadError={loadError}
+        />
+      );
+      return [alertsTab, detailsTab, resourcesTab, observeTab];
+    }, [resource, alerts, loaded, loadError]);
+
+  const tabs: TabPage[] = [
+    {
+      title: t('Alerts'),
+      href: '#sidebar-alerts',
+      component: () => alertsTab,
+    },
+    {
+      title: t('Details'),
+      href: '#sidebar-details',
+      component: () => detailsTab,
+    },
+    {
+      title: t('Resources'),
+      href: '#sidebar-resources',
+      component: () => resourcesTab,
+    },
+    {
+      title: t('Observe'),
+      href: '#sidebar-observe',
+      component: () => observeTab,
+    },
+  ];
+
+  const title = <DetailsPageTitle resource={resource} resourceModel={model} />;
+
+  return (
+    <>
+      <PageHeading
+        title={title}
+        resource={resource}
+        getResourceStatus={getResourceStatus}
+        className="odf-topology-sidebar-heading"
+      />
+      {tabs.length > 0 ? (
```

Comment:
"tabs.length > 0" will this ever be false ??
there will always be something in "tabs" list...

Reply:
u can use "inFlight" instead...
 ```suggestion
      {!inFlight ? (
```

Reply:
while using "useMemo" on "tabs"... make sure"
`!inFlight && loaded && !loadError ? TabPage[] : [];`

Reply:
now after above condition, `tabs.length > 0` this should work...